crates/codegen/schema/src/types/parser.rs

```diff
@@ -36,22 +38,25 @@ pub enum ParserDefinition {
 
     #[schemars(title = "Repeat Expression")]
     Repeat {
+        expression: ParserRef,
         min: usize,
         max: usize,
-        expression: ParserRef,
     },
 
     #[schemars(title = "SeparatedBy Expression")]
     SeparatedBy {
-        separator: String,
         expression: ParserRef,
+        separator: Reference,
```

Comment:
Not sure why a `Reference` is used here, rather than a regular `ParserRef`. Is there a parser limitation?